packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -383,15 +389,16 @@ class DateTimePicker extends React.Component {
          * Reference https://github.com/cerner/terra-framework/issues/1086
          */
         if (this.hourInput) {
+          this.isDateSelected = false;
```

Comment:
Can you explain what this is used for here and the check on line 383?

Reply:
The condition in question here was added to overcome the issue when a date is selected from the dropdown picker, the focus was not moving to the time input (even though the value was valid).
This issue arose because of the condition `event.currentTarget === this.dateInputComponent` which was added to help move focus when the date was entered manually.
So by adding the `isDateSelected` condition resolved the focus issue.